### ISSUE TITLE
A way to "delete" a namespace from NG

### ISSUE THREAD

**Issue by carols10cents** [influxdata/influxdb_iox#4467]
As a customer, I would like to be able to: 

- write some data into the NG namespace `myorg_mybucket`
- decide oh wait no, that data is all crap, delete the namespace `myorg_mybucket`
- then write new data into `myorg_mybucket`
- do a query and only get back the data written since the delete request

**Comment by alamb** [comment]
> I'm assuming for now that this should be done as a "soft" deletion? That is, add a "deleted" column to the namespaces table in the catalog, scope all other namespace queries to be deleted = 'f', and only flip that column when a namespace is requested to be deleted

Yeah, that seems like a reasonable place to start (and then we can eventually implement a `hard delete` operation that actually drops the entries from the catalog (maybe the postgres database will just automatically cascade if the relevant schema object is dropped due to the definition of `foreign keys`

I think:
1. @pauldix  should weigh in on this design discussion
2. We should evaluate if dropping a database is required for #4412 or if we can just stub the API out and return "not implemented yet" to unblock forward progress
